### ISSUE TITLE
RadListview pullToRefresh function to Clear all data  error occur

### ISSUE THREAD

**Issue by yclau** [telerik/nativescript-ui-feedback#643]
### Tell us about the problem
When I pullToRefresh after clear all data using splice  error occur below shown.

```
Fatal JavaScript exception - application has been terminated. (/Users/abc/Documents/Project/StockTips/node_modules/nativescript-ui-listview/ui-listview.ios.js:1408:53)
Error: UICollectionView received layout attributes for a cell with an index path that does not exist: <NSIndexPath: 0xc000000000000016> {length = 2, path = 0 - 0}
file:///app/tns_modules/nativescript-ui-listview/ui-listview.js:1408:53: JS ERROR Error: UICollectionView received layout attributes for a cell with an index path that does not exist: <NSIndexPath: 0xc000000000000016> {length = 2, path = 0 - 0}
```

### Which platform(s) does your issue occur on?
_iOS 

### Please provide the following version numbers that your issue occurs with:
```
"nativescript": {
		"id": "com.yctech.stocktips",
		"tns-android": {
			"version": "4.0.1"
		},
		"tns-ios": {
			"version": "4.0.1"
		}
	},
	"dependencies": {
		"nativescript-iqkeyboardmanager": "^1.3.0",
		"nativescript-loading-indicator": "^2.4.0",
		"nativescript-modal-datetimepicker": "^1.1.3",
		"nativescript-plugin-firebase": "^5.3.1",
		"nativescript-social-share": "^1.5.0",
		"nativescript-theme-core": "^1.0.4",
		"nativescript-ui-listview": "^3.5.5",
		"tns-core-modules": "^4.0.0"
	},
```

**Comment by NickIliev** [comment]
@yclau can you provide a sample project or [Playground demo](https://play.nativescript.org/) that can reproduce the issue

**Comment by bradmartin** [comment]
Ran into this error recently also. Haven't quite pinned it down, will update when I have an idea of what's happening. My scenario is slightly different. I have a RLV that I have a 'filter' input that I query a server for data on text change. When I actually get a match and add the items to the ObservableArray and then disable the `loadOnDemandMode` it throws.

**Comment by lini** [comment]
A fix for using splice on the observable array source of the listview was published in the latest release v3.5.8 earlier this week. Can you give it a try and check if it works for you?
